docs/ref/models/conditional-expressions.txt

```diff
@@ -236,3 +241,28 @@ On other databases, this is emulated using a ``CASE`` statement:
 
 The two SQL statements are functionally equivalent but the more explicit
 ``FILTER`` may perform better.
+
+Conditional filter
+------------------
+
+.. versionadded:: 2.1
+
+When a conditional expression returns a boolean value, it is possible to
+use the expression directly in a filter expression. This means it will not
+be returned as a column in the query, but will still be used to filter
+results::
+
+    >>> non_unique_account_type = Client.objects.exclude(pk=OuterRef('pk')).filter(
+    ...     account_type=OuterRef('account_type'))
```

Comment:
You might want to perform the `filter` by `account_type` before the `exclude(pk)` to match the order in which the inner queries criterias are defined in the SQL block below.